build-tools/automation/build.groovy

```diff
@@ -12,93 +10,37 @@ def hasPrLabelFullMonoIntegrationBuild = false
-    try {
-        stage(stageName) {
+def utils = null
+
+prLabels = null  // Globally defined "static" list accessible within the hasPrLabel function
+
+timestamps {
+    node("${env.BotLabel}") {
+        def scmVars = null
+
+        stage ("checkout") {
+            def ctAttempts = 3
             def retryAttempt = 0
             def waitSecondsBeforeRetry = 15
             retry(ctAttempts) {     // Retry will always invoke the body at least once for an attempt count of 0 or 1
-                timeout(time: timeoutValue, unit: timeoutUnit) {
-                    dir(directory) {
-                        if (retryAttempt > 0) {
-                            echo "WARNING : Stage ${stageName} failed on try #${retryAttempt}. Waiting ${waitSecondsBeforeRetry} seconds"
-                            sleep(waitSecondsBeforeRetry)
-                            echo "Retrying ..."
-                            waitSecondsBeforeRetry = waitSecondsBeforeRetry * 2
-                        }
-
-                        retryAttempt++
-                        body()
+                dir (XADir) {
+                    if (retryAttempt > 0) {
+                        echo "WARNING : Stage checkout failed on try #${retryAttempt}. Waiting ${waitSecondsBeforeRetry} seconds"
+                        sleep(waitSecondsBeforeRetry)
+                        echo "Retrying ..."
+                        waitSecondsBeforeRetry = waitSecondsBeforeRetry * 2
                     }
+
+                    retryAttempt++
+                    scmVars = checkout scm
                 }
             }
         }
-    } catch (error) {
-        def result = fatal ? 'ERROR' : 'UNSTABLE'
-        echo "ERROR : ${stageName}: Unexpected error: ${error}. Marking build as ${result}."
-        currentBuild.result = result
-        if (fatal) {
-            throw error
-        }
-    } finally {
-        echo "Stage result: ${stageName}: ${currentBuild.currentResult}"
-    }
-}
 
-def publishPackages(filePaths) {
-    def status = 0
-    try {
-         // Note: The following function is provided by the Azure Blob Jenkins plugin
-         azureUpload(storageCredentialId: env.StorageCredentialId,
-                 storageType: "blobstorage",
-                 containerName: env.ContainerName,
-                 virtualPath: env.StorageVirtualPath,
-                 filesPath: filePaths,
-                 allowAnonymousAccess: true,
-                 pubAccessible: true,
-                 doNotWaitForPreviousBuild: true,
-                 uploadArtifactsOnlyIfSuccessful: true)
-    } catch (error) {
-        echo "ERROR : publishPackages: Unexpected error: ${error}"
-        status = 1
-    }
-
-    return status
-}
-
-prLabels = null  // Globally defined "static" list accessible within the hasPrLabel function
+        def workspace = "/Users/${env.USER}/jenkins/workspace/${env.JOB_BASE_NAME}"
+        utils = load "${workspace}/${XADir}/build-tools/automation/utils.groovy"
```

Comment:
Why do you need the workspace here? does using a relative path not work with `load` ?
If we really need it we shouldn't hardcode the path but use `env.WORKSPACE` instead.

Reply:
Relative path indeed works.  Addressed with commit https://github.com/xamarin/xamarin-android/pull/2838/commits/316854a69cb39775d136433e566b4307a3ffe9c5